src/battle_main.c

```diff
@@ -5995,9 +5995,52 @@ void RunBattleScriptCommands(void)
         gBattleScriptingCommandsTable[gBattlescriptCurrInstr[0]]();
 }
 
+bool32 TrySetAteType(u32 move, u32 battlerAtk, u32 attackerAbility)
+{
+    u32 ateType;
+    u32 moveEffect = gMovesInfo[move].effect;
+
+    if (moveEffect == EFFECT_HIDDEN_POWER
+     || moveEffect == EFFECT_WEATHER_BALL
+     || moveEffect == EFFECT_CHANGE_TYPE_ON_ITEM
+     || moveEffect == EFFECT_NATURAL_GIFT
+     || (moveEffect == EFFECT_TERA_BLAST && IsTerastallized(battlerAtk))
+     || (moveEffect == EFFECT_TERA_STARSTORM && gBattleMons[battlerAtk].species == SPECIES_TERAPAGOS_STELLAR))
+        return FALSE;
```

Comment:
Couldn't we change some of this check to refer to a field in `gBattleMoveEffects` instead? The Tera stuff is obviously not doable that way, but gathering gathering information in a single place is probably beneficial

Reply:
We could but I personally don't like using a bit for a couple of checks.